.github/workflows/py-workflow.yml

```diff
@@ -54,6 +54,16 @@ jobs:
     - name: install build-requirements
       run: python3 -m pip install -r build-requirements.txt
 
+    - name: Check if pre-commit config exists
+      id: check_files
+      uses: andstor/file-existence-action@v1
+      with:
+        files: ".pre-commit-config.yaml"
+
+    - name: Run pre-commit
+      if: steps.check_files.outputs.files_exists == 'true'
+      uses: pre-commit/action@v3.0.0
```

Comment:
notes from the maintainer:

> DEPRECATED this action is in maintenance-only mode and will not be accepting new features.
> 
> Please switch to using [pre-commit.ci](https://pre-commit.ci/) which is faster and has more features.

Reply:
Yeah, the action is so simple that if needed we can even copy it and lose the dependency. Also pre-commit.ci is being run a SAAS, which, at least for me doesn't make much sense.